analyses/cell-type-consensus/exploratory-notebooks/utils/setup-functions.R

```diff
@@ -15,7 +15,7 @@ summarize_celltypes <- function(file, id){
     dplyr::ungroup()
   
   summary_df <- df |> 
-    dplyr::group_by(library_id, sample_type, consensus_annotation, consensus_ontology) |> 
+    dplyr::group_by(project_id, library_id, sample_type, consensus_annotation, consensus_ontology, num_celltypes) |> 
     dplyr::summarize(total_cells_per_annotation = dplyr::n(),
                      total_cells_per_library = unique(total_cells_per_library)) |>
```

Comment:
I'm assuming `num_celltypes` is constant per library, so I would probably not put it into the grouping list. 
```suggestion
    dplyr::group_by(project_id, library_id, sample_type, consensus_annotation, consensus_ontology) |> 
    dplyr::summarize(
      num_celltypes = unique(num_celltypes), # constant for each library
      total_cells_per_annotation = dplyr::n(),
      total_cells_per_library = unique(total_cells_per_library) # constant for each library
    ) |>
```

Or put `total_cells_per_library` into the grouping as well? I just feel like the treatment should be consistent.